tests/Unit/Services/KrScoringServiceTest.php

```
@@ -107,4 +107,80 @@ public function testScoreForMergedPrWithGeminiError(): void
         $service->scoreForMergedPr('https://github.com/org/repo/pull/789', 1);
         $this->assertTrue(true);
     }
+
+    public function testScoreForMergedPrBulkLoadsAndDeduplicatesKeyResults(): void
+    {
+        $links = [
+            ['id' => 11, 'local_type' => 'hl_work_item', 'local_id' => 10, 'ref_label' => 'PR #10'],
+            ['id' => 12, 'local_type' => 'hl_work_item', 'local_id' => 10, 'ref_label' => 'PR #10'],
+        ];
+        $kr = [
```

PHPMD (2.15.0)

[warning] Avoid variables with short names like $kr. Configured minimum length is 3. (undefined)

(ShortVariable)